code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -335,4 +335,142 @@ var/global/chicken_count = 0
+			T = O
+
+		if(QUALITY_WELDING in T.tool_qualities)
+			if(health < maxHealth)
+				if(T.use_tool(user, src, WORKTIME_NORMAL, QUALITY_WELDING, FAILCHANCE_EASY, required_stat = STAT_MEC))
+					health = maxHealth
+					to_chat(user, "You repair the damage to [src].")
+					return
+				return
+			to_chat(user, "[src] doesn't need repairs.")
+			return
+
+	if(istype(O, /obj/item/reagent_containers/food)) //feedin' dem chickens
+		var/obj/item/reagent_containers/food/snacks/grown/G = O
+		if(G.reagents.total_volume >= reagents.maximum_volume - reagents.total_volume)
+			to_chat(user, "\blue [name] is completly full!")
+			return
+		user.visible_message("\blue [user] feeds [O] to [name]! She clucks happily.","\blue You feed [O] to [name]! She clucks happily.")
+		user.drop_item()
+		G.reagents.trans_to_mob(src, G.reagents.total_volume, CHEM_BLOOD)
+		qdel(O)
+	else
+		..()
+
+/mob/living/simple_animal/metal_chicken/Life()
+	. =..()
+	if(!.)
+		return
+	if(!stat && prob(50) && lay_egg())
+		visible_message("[src] [pick("lays an egg.","squats down and croons.","begins making a huge racket.","begins clucking raucously.")]")
+
+/mob/living/simple_animal/metal_chicken/proc/lay_egg()
+	if(!reagents.reagent_list.len)
+		return FALSE
+	var/chem = pick(reagents.reagent_list).id
+	var/egg_type
+	if(reagents.remove_reagent(chem, chem_to_egg, TRUE))
+		switch(chem)
+			// Turn liquid metal into solids
+			if(MATERIAL_IRON) // Steel isn't a chem, so we use the next best thing
+				egg_type = /obj/item/stack/material/steel
+			if(MATERIAL_SILVER)
+				egg_type = /obj/item/stack/material/silver
+			if(MATERIAL_GOLD)
+				egg_type = /obj/item/stack/material/gold
+			if(MATERIAL_PLASMA)
+				egg_type = /obj/item/stack/material/plasma
+			if(MATERIAL_URANIUM)
+				egg_type = /obj/item/stack/material/uranium
+			if("plasticide")
+				egg_type = /obj/item/stack/material/plastic
+			if("woodpulp")
+				egg_type = /obj/item/stack/material/wood
+			if("crayon_dust")
+				egg_type = /obj/random/rations/crayon
+			if("slimejelly")
+				egg_type = /obj/item/slime_extract/grey
+			if("milk")
+				egg_type = /obj/item/reagent_containers/food/snacks/sliceable/cheesewheel
+			if("silicate")
+				egg_type = /obj/item/stack/material/glass
+			if("nicotine")
+				egg_type = /obj/item/clothing/mask/smokable/cigarette
+			if("pararein")
+				egg_type = /obj/effect/spider/spiderling
+			if("blattadin ")
+				egg_type = /obj/item/reagent_containers/food/snacks/cube/roach
+			else // Empty bottle that then get filled with left-over chems
+				egg_type = /obj/item/reagent_containers/glass/bottle
+
+	if(!egg_type) // In case we don't have something to lay.
+		return FALSE
+
+	var/egg = new egg_type(get_turf(src))
+	if(istype(egg, /obj/item/reagent_containers))
+		var/obj/item/reagent_containers/RC = egg
+		RC.reagents.clear_reagents() // Remove all of the previous chem, just in case
+		RC.reagents.add_reagent(chem, chem_to_egg) // Add the reagent in quantity it is supposed to have.
```

Comment:
Do all the possible objects for egg type need to have a reagent added to them?

Reply:
Only for those that don't spawn anything specific.
If they don't, then it choose to spawn a bottle, filled with the 20 chems that it used up.